src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -278,12 +315,20 @@ public void grantWorkspaceRole(
         samActionToModifyRole(role));
     ResourcesApi resourceApi = samResourcesApi(userReq.getRequiredToken());
     try {
-      resourceApi.addUserToPolicy(
-          SamConstants.SAM_WORKSPACE_RESOURCE, workspaceId.toString(), role.toSamRole(), email);
+      SamRetry.retry(
+          () ->
+              resourceApi.addUserToPolicy(
+                  SamConstants.SAM_WORKSPACE_RESOURCE,
+                  workspaceId.toString(),
+                  role.toSamRole(),
+                  email));
       logger.info(
           "Granted role {} to user {} in workspace {}", role.toSamRole(), email, workspaceId);
     } catch (ApiException apiException) {
       throw SamExceptionFactory.create("Error granting workspace role in Sam", apiException);
+    } catch (InterruptedException interruptedException) {
+      throw SamExceptionFactory.create(
```

Comment:
But these other methods aren't `Wrapped`? Seems error prone that we may start to use this in a step. I think if we really want to avoid rethrowing InterruptedExceptions in Steps, we need to make it default to throw the InterruptedException. Or if it doesn't matter when we don't throw the InterruptedException, then we don't bother.